ControlFlowArraysAndStrings/ControlFlowArraysAndStrings/Program.cs

```diff
@@ -0,0 +1,343 @@
+        }
+
+        /// <summary>
+        /// Removes the value at the beginning of a specified array.
+        /// </summary>
+        /// <param name="array">The array to be changed.</param>
+        /// <returns>A new array with a value removed from the start.</returns>
+        public static string[] RemoveElementFromStart(string[] array)
+        {
+            string[] newArray = new string[array.Length - 1];
+
+            for (int i = 1; i < array.Length; i++)
+            {
+                newArray[i - 1] = array[i];
+            }
+
+            return newArray;
+        }
+
+        /// <summary>
+        /// Removes the value at the end of the specified array.
+        /// </summary>
+        /// <param name="array">The array to be changed.</param>
+        /// <returns>A new array with a value removed from the end.</returns>
+        public static string[] RemoveElementFromEnd(string[] array)
+        {
+            string[] newArray = new string[array.Length - 1];
+
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                newArray[i] = array[i];
+            }
+
+            return newArray;
+        }
+
+        /// <summary>
+        /// Removes the value at the specified index from the specified array.
+        /// </summary>
+        /// <param name="array">The array to be changed.</param>
+        /// <param name="index">The index to be removed.</param>
+        /// <returns>A new array with a value removed.</returns>
+        public static string[] RemoveElementFromAnywhere(string[] array, int index)
+        {
+            string[] newArray = new string[array.Length - 1];
+
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                if (i < index)
+                {
+                    newArray[i] = array[i];
+                }
+
+                if (i >= index)
+                {
+                    newArray[i] = array[i + 1];
+                }
+            }
+
+            return newArray;
+        }
+
+        /// <summary>
+        /// Attempts to log in a user and prints a message to the console if login 
+        /// was successful or unsuccessful. Allows 3 unsuccessful attempts before lock out.
+        /// </summary>
+        public static void Login()
+        {
+            int loginAttempts = 3;
+            string[] credentialType = { "user name", "password"};
+            bool[] isValid = { false, false };
+
+            for (int i = 0; i < credentialType.Length; i++)
+            {
+                while (!isValid[i] && loginAttempts != 0)
+                {
+                    string credential = GetLoginCredentialsFromUser(credentialType[i]);
+                    isValid[i] = ValidateCredential(credentialType[i], credential);
+
+                    if (!isValid[i])
+                    {
+                        loginAttempts--;
+                        Console.WriteLine($"Invalid {credentialType[i]}. {loginAttempts} tries left.");
```

Comment:
I'm not running this to test but if I try entering a bad username it looks like it might still prompt me for a password?  Also, not really part of the assignment, but if you tell me as soon as I enter the username then it is technically a security issue because you give away which usernames are valid and which are not.

Reply:
It prompts for a username until a valid one is entered or `loginAttempts` hits 0. If the attempts are at 0, it'll break out of the outer loop and "lock" them out, they won't get prompted for a password. If they haven't run out of attempts and the username is valid it will prompt for a password.